### PR TITLE
Implement the CLI

### PR DESCRIPTION
**Pull request by huertas**

### CODE REVIEW

File: powerapi-cli/src/main/scala/org/powerapi/app/Application.scala [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+  def printHelp(): Unit = {
+    val str =
+      """
+        |PowerAPI (c) Spirals Team"
+        |
+        |Build a software-defined power meter. Don't forget to configure correctly the modules (see the documentation).
+        |
+        |usage: ./powerapi-cli modules [cpu-simple|cpu-dvfs|libpfm-core|libpfm-core-proces|powerspy, ...]
+        |                              monitor --frequency [ms] --targets [pid, ..., app, ...)|all] --agg [max|min|geomean|logsum|mean|median|stdev|sum|variance] --[console,file [filepath],chart]
```
Comment (rouvoy) [inline powerapi-cli/src/main/scala/org/powerapi/app/Application.scala]: What does "monitor" stands for?

File: powerapi-cli/src/main/scala/org/powerapi/app/Application.scala [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+        |                      duration [s]
+        |
+        |example: ./powerapi-cli modules cpu-simple monitor --frequency 1000 --targets firefox --agg max --console monitor --targets chrome --agg max --console
+        |                        modules powerspy monitor --frequency 1000 --targets all -agg max --console
```
Comment (rouvoy) [inline powerapi-cli/src/main/scala/org/powerapi/app/Application.scala]: Fix: `-agg` with `--agg`

File: powerapi-cli/src/main/scala/org/powerapi/app/Application.scala [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+  }
+
+  def printHelp(): Unit = {
+    val str =
+      """
+        |PowerAPI (c) Spirals Team"
+        |
+        |Build a software-defined power meter. Don't forget to configure correctly the modules (see the documentation).
+        |
+        |usage: ./powerapi-cli modules [cpu-simple|cpu-dvfs|libpfm-core|libpfm-core-proces|powerspy, ...]
```
Comment (rouvoy) [inline powerapi-cli/src/main/scala/org/powerapi/app/Application.scala]: Fix: `modules` with `--module`

File: powerapi-cli/src/main/scala/org/powerapi/app/Application.scala [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+      }
+    }).toSeq
+  }
+
+  def printHelp(): Unit = {
+    val str =
+      """
+        |PowerAPI (c) Spirals Team"
+        |
+        |Build a software-defined power meter. Don't forget to configure correctly the modules (see the documentation).
```
Comment (rouvoy) [inline powerapi-cli/src/main/scala/org/powerapi/app/Application.scala]: Replace `Don't` by `Do not`

File: powerapi-cli/src/main/scala/org/powerapi/app/Application.scala [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+        case pidR(pid) => Process(pid.toInt)
+        case appR(app) => Application(app)
+      }
+    }).toSeq
+  }
+
+  def printHelp(): Unit = {
+    val str =
+      """
+        |PowerAPI (c) Spirals Team"
```
Comment (rouvoy) [inline powerapi-cli/src/main/scala/org/powerapi/app/Application.scala]: PowerAPI is not copyrighted ;)

File: powerapi-cli/src/main/scala/org/powerapi/app/Application.scala [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+        |
+        |Build a software-defined power meter. Don't forget to configure correctly the modules (see the documentation).
+        |
+        |usage: ./powerapi-cli modules [cpu-simple|cpu-dvfs|libpfm-core|libpfm-core-proces|powerspy, ...]
+        |                              monitor --frequency [ms] --targets [pid, ..., app, ...)|all] --agg [max|min|geomean|logsum|mean|median|stdev|sum|variance] --[console,file [filepath],chart]
+        |                      duration [s]
+        |
+        |example: ./powerapi-cli modules cpu-simple monitor --frequency 1000 --targets firefox --agg max --console monitor --targets chrome --agg max --console
+        |                        modules powerspy monitor --frequency 1000 --targets all -agg max --console
+        |                        duration 30
```
Comment (rouvoy) [inline powerapi-cli/src/main/scala/org/powerapi/app/Application.scala]: Does this refer to the above line?

File: powerapi-cli/src/universal/scripts/system.bash [hunk 7]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+ulimit -n 4096
+
+exit
```
Comment (rouvoy) [inline powerapi-cli/src/universal/scripts/system.bash]: Missing empty line

File: powerapi-cli/src/main/scala/org/powerapi/app/Application.scala (hunk 1 above)
Comment (mcolmant) [inline powerapi-cli/src/main/scala/org/powerapi/app/Application.scala]: In order to start a new monitor (see the example). Useful for creating several monitorings on a same PowerMeter.

File: powerapi-cli/src/main/scala/org/powerapi/app/Application.scala (hunk 3 above)
Comment (mcolmant) [inline powerapi-cli/src/main/scala/org/powerapi/app/Application.scala]: I saw that like an entry (as duration), thus we can use the option with -- after. It is possible to plug different modules also on the same PowerMeter.

File: powerapi-cli/src/main/scala/org/powerapi/app/Application.scala (hunk 6 above)
Comment (mcolmant) [inline powerapi-cli/src/main/scala/org/powerapi/app/Application.scala]: Yes, it is an example of use. Here it creates two PowerMeter. One with cpu-simple, and another one with powerspy. Two monitorings will be attached to the first one, one for the second.
The application will be run during 30 seconds (for all PowerMeter)

File: powerapi-cli/src/main/scala/org/powerapi/app/Application.scala [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+  val aggR = """max|min|geomean|logsum|mean|median|stdev|sum|variance""".r
+  val durationR = """\d+""".r
+  val pidR = """(\d+)""".r
+  val appR = """(.+)""".r
+
+  @volatile var powerMeters = Seq[PowerMeter]()
+  @volatile var monitors = Seq[PowerMonitoring]()
+
+  val shutdownHookThread = scala.sys.ShutdownHookThread {
+    println("It's the time for sleeping! ...")
```
Comment (rouvoy) [inline powerapi-cli/src/main/scala/org/powerapi/app/Application.scala]: Maybe we should avoid such messages.

File: powerapi-cli/src/main/scala/org/powerapi/app/Application.scala [hunk 9]
```diff
@@ -0,0 +1,218 @@
...
+      """
+        |PowerAPI, Spirals Team"
+        |
+        |Build a software-defined power meter. Do not forget to configure correctly the modules (see the documentation).
+        |
+        |usage: ./powerapi-cli modules [cpu-simple|cpu-dvfs|libpfm-core|libpfm-core-proces|powerspy, ...]
+        |                              monitor --frequency [ms] --targets [pid, ..., app, ...)|all] --agg [max|min|geomean|logsum|mean|median|stdev|sum|variance] --[console,file [filepath],chart]
+        |                      duration [s]
+        |
+        |example: ./powerapi-cli modules cpu-simple monitor --frequency 1000 --targets firefox --agg max --console monitor --targets chrome --agg max --console
```
Comment (rouvoy) [inline powerapi-cli/src/main/scala/org/powerapi/app/Application.scala]: add a `\` if the line is not ended.

File: powerapi-cli/src/main/scala/org/powerapi/app/Application.scala [hunk 10]
```diff
@@ -0,0 +1,218 @@
...
+  }
+
+  def printHelp(): Unit = {
+    val str =
+      """
+        |PowerAPI, Spirals Team"
+        |
+        |Build a software-defined power meter. Do not forget to configure correctly the modules (see the documentation).
+        |
+        |usage: ./powerapi-cli modules [cpu-simple|cpu-dvfs|libpfm-core|libpfm-core-proces|powerspy, ...]
```
Comment (rouvoy) [inline powerapi-cli/src/main/scala/org/powerapi/app/Application.scala]: Why don't we call it `powerapi` instead of `powerapi-cli`?

**Pull request merged**